packages/airqo-device-analysis/airqloudanalysis/airqoairqloudanalysis/datainitialisation.py

```
@@ -0,0 +1,740 @@
+def time_last_post(df):
+    results = []
+    for index, row in df.iterrows():
+        device_number = row[device_number_literal]
+        read_key = row[read_key_literal_1]
+        device_id = row[device_id_literal_1]
+        airqloud = row[airqloud_number_literal]
+        last = last_url(device_id, read_key)
+
+        # Fetch data from the API
+        try:
+            last_data = requests.get(last)
```

Ruff (0.15.12)

[error] Probable use of `requests` call without timeout

(S113)